Purchases/Public/RCPurchases.m

```diff
@@ -957,23 +957,42 @@ - (void)handlePurchasedTransaction:(SKPaymentTransaction *)transaction
                                       [self.presentedOfferingsByProductIdentifier removeObjectForKey:productIdentifier];
                                   }
 
-                              [self.backend postReceiptData:data
-                                                  appUserID:self.appUserID
-                                                  isRestore:[self allowSharingAppStoreAccount]
-                                          productIdentifier:productIdentifier
-                                                      price:price
-                                                paymentMode:paymentMode
-                                          introductoryPrice:introPrice
-                                               currencyCode:currencyCode
-                                          subscriptionGroup:subscriptionGroup
-                                                  discounts:discounts
-                                presentedOfferingIdentifier:presentedOffering
-                                                 completion:^(RCPurchaserInfo *_Nullable info,
-                                                         NSError *_Nullable error) {
-                                                     [self handleReceiptPostWithTransaction:transaction
-                                                                              purchaserInfo:info
-                                                                                      error:error];
-                                                 }];
+                                  [self.backend postReceiptData:data
+                                                      appUserID:self.appUserID
+                                                      isRestore:self.allowSharingAppStoreAccount
+                                              productIdentifier:productIdentifier
+                                                          price:price
+                                                    paymentMode:paymentMode
+                                              introductoryPrice:introPrice
+                                                   currencyCode:currencyCode
+                                              subscriptionGroup:subscriptionGroup
+                                                      discounts:discounts
+                                    presentedOfferingIdentifier:presentedOffering
```

Comment:
for another PR but methods with more than ~5/6 parameters should be refactored. Easy trick is to create a meta object that contains grouped params.

Reply:
Totally Agree